### DISCUSSION TITLE
Running `postUpgradeTasks` on file inside directory

### DISCUSSION THREAD

**Discussion by cschmatzler** [renovatebot/renovate#22862] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovatebot/github-action@v38.1.7

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I am trying to run some post upgrade tasks after upgrades, that depend on the directory that the file is in.

### What you tried so far.

Assume we have this file at `clusters/lib/cilium/chartfile.yaml`:
```
version: 1
directory: charts
repositories:
  - name: cilium
    url: https://helm.cilium.io/
requires:
  - chart: cilium/cilium
    # renovate: datasource=helm depName=cilium registryUrl=https://helm.cilium.io versioning=helm
    version: 1.13.4
```
and this regex manager for it:
```
"regexManagers": [
    {
      "fileMatch": [
        "(^|/)chartfile\\.yaml$"
      ],
      "matchStrings": [
        "renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?) registryUrl=(?<registryUrl>.*) versioning=(?<versioning>.*?)\\s*version: (?<currentValue>.*)\\s*"
      ],
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
    }
  ],
```

That works! I am getting PRs with the correct new version. Now, the way Tanka chartfiles work, just changing the version won't actually upgrade anything, since the Helm chart itself needs to be vendored into the repository.
For this, I have tried the following:
```
"packageRules": [
{
      "matchFiles": [
        "(^|/)chartfile\\.yaml$"
      ],
      "postUpgradeTasks": {
        "commands": [
          "curl -sL -o- https://get.helm.sh/helm-v$(awk '/^helm / { print $2 }' .tool-versions)-linux-amd64.tar.gz | tar xz --strip-components=1 -C /usr/local/bin/ linux-amd64/helm && chmod +x /usr/local/bin/helm",
          "curl -sL -o /usr/local/bin/tk https://github.com/grafana/tanka/releases/download/v$(awk '/^tanka / { print $2 }' .tool-versions)/tk-linux-amd64 && chmod +x /usr/local/bin/tk",
          "tk tool charts vendor"
        ],
        "fileFilters": [
          "chartfile.yaml",
          "charts/**"
        ]
      }
    }
]
```

The idea is that I install the Tanka and Helm CLIs and vendor the updated chart into the `charts/` directory, then commit that. The problem is that this doesn't work, presumably because the `postUpgradeTasks` commands are run into the wrong directory? I am actually not sure. The result I want to achieve is that on matching `chartfile.yaml`, Renovate changes working directory into whatever directory that chartfile is in, and then runs `tk tool charts vendor` after installing dependencies. I have, so far, not found a way to change the working directory like that.

Any help would be much appreciated :)

### Relevant debug logs

The post upgrade commands are not showing up in the logs at all.

**Comment by rarkins** [comment, marked as answer]
`matchFiles` needs an exact match, not regex

**Reply by cschmatzler** [reply]
Ah. Consolidating all the charts into one chartfile and referencing that specifically works. Thanks so much!
